### ISSUE TITLE
SMTP Mail Issue

### ISSUE THREAD

**Issue by tyler2k15** [swiftmailer/swiftmailer#1009]
| Q                   | A
| ------------------- | -----
| Bug report?         | yes
| Feature request?    | no
| RFC?                | no
| How used?           | 3rd party - pterodactyl
| Swiftmailer version | 5.4.8
| PHP version         | `PHP 7.1.10-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Sep 29 2017 17:04:25) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.10-1+ubuntu16.04.1+deb.sury.org+1, Copyright (c) 1999-2017, by Zend Technologies`

### Observed behaviour
Fails to send SMTP emails on my Pterodactyl installation

### Expected behaviour
Send emails without error

### Example
`failed_jobs` database records the following two errors: 

**Note: Same exact settings work on same machine for Xenforo installation SMTP mail sending**

```
Swift_TransportException: Expected response code 220 but got code "", with message "" in /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php:383
Stack trace:
#0 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(289): Swift_Transport_AbstractSmtpTransport->_assertResponseCode('', Array)
#1 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(117): Swift_Transport_AbstractSmtpTransport->_readGreeting()
#2 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mailer.php(79): Swift_Transport_AbstractSmtpTransport->start()
#3 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(440): Swift_Mailer->send(Object(Swift_Message), Array)
#4 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(211): Illuminate\Mail\Mailer->sendSwiftMessage(Object(Swift_Message))
#5 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/Channels/MailChannel.php(60): Illuminate\Mail\Mailer->send(Object(Illuminate\Support\HtmlString), Array, Object(Closure))
#6 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(113): Illuminate\Notifications\Channels\MailChannel->send(Object(Pterodactyl\Models\User), Object(Pterodactyl\Notifications\SendPasswordReset))
#7 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(89): Illuminate\Notifications\NotificationSender->sendToNotifiable(Object(Pterodactyl\Models\User), '33fa6d1f-21eb-4...', Object(Pterodactyl\Notifications\SendPasswordReset), 'mail')
#8 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/ChannelManager.php(51): Illuminate\Notifications\NotificationSender->sendNow(Object(Illuminate\Database\Eloquent\Collection), Object(Pterodactyl\Notifications\SendPasswordReset), Array)
#9 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/SendQueuedNotifications.php(57): Illuminate\Notifications\ChannelManager->sendNow(Object(Illuminate\Database\Eloquent\Collection), Object(Pterodactyl\Notifications\SendPasswordReset), Array)
#10 [internal function]: Illuminate\Notifications\SendQueuedNotifications->handle(Object(Illuminate\Notifications\ChannelManager))
#11 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#12 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#13 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#14 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#15 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(94): Illuminate\Container\Container->call(Array)
#16 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(114): Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}(Object(Illuminate\Notifications\SendQueuedNotifications))
#17 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Notifications\SendQueuedNotifications))
#18 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(98): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#19 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(42): Illuminate\Bus\Dispatcher->dispatchNow(Object(Illuminate\Notifications\SendQueuedNotifications), false)
#20 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(69): Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\DatabaseJob), Array)
#21 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(317): Illuminate\Queue\Jobs\Job->fire()
#22 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(267): Illuminate\Queue\Worker->process('database', Object(Illuminate\Queue\Jobs\DatabaseJob), Object(Illuminate\Queue\WorkerOptions))
#23 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(113): Illuminate\Queue\Worker->runJob(Object(Illuminate\Queue\Jobs\DatabaseJob), 'database', Object(Illuminate\Queue\WorkerOptions))
#24 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\Queue\Worker->daemon('database', 'high,standard,l...', Object(Illuminate\Queue\WorkerOptions))
#25 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(85): Illuminate\Queue\Console\WorkCommand->runWorker('database', 'high,standard,l...')
#26 [internal function]: Illuminate\Queue\Console\WorkCommand->fire()
#27 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#28 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#29 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#30 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#31 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\Container\Container->call(Array)
#32 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Command/Command.php(264): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#33 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Console/Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#34 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Application.php(869): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#35 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Application.php(223): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#36 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Application.php(130): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#37 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#38 /var/www/html/mcadmin.phoenixden.net/public_html/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#39 {main}
```

Error 2:

```
Swift_TransportException: Connection to tcp://send.one.com:465 Timed Out in /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php:404
Stack trace:
#0 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(289): Swift_Transport_AbstractSmtpTransport->_getFullResponse(0)
#1 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(117): Swift_Transport_AbstractSmtpTransport->_readGreeting()
#2 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mailer.php(79): Swift_Transport_AbstractSmtpTransport->start()
#3 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(440): Swift_Mailer->send(Object(Swift_Message), Array)
#4 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(211): Illuminate\Mail\Mailer->sendSwiftMessage(Object(Swift_Message))
#5 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/Channels/MailChannel.php(60): Illuminate\Mail\Mailer->send(Object(Illuminate\Support\HtmlString), Array, Object(Closure))
#6 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(113): Illuminate\Notifications\Channels\MailChannel->send(Object(Pterodactyl\Models\User), Object(Pterodactyl\Notifications\SendPasswordReset))
#7 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(89): Illuminate\Notifications\NotificationSender->sendToNotifiable(Object(Pterodactyl\Models\User), '83c4ba63-a4f3-4...', Object(Pterodactyl\Notifications\SendPasswordReset), 'mail')
#8 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/ChannelManager.php(51): Illuminate\Notifications\NotificationSender->sendNow(Object(Illuminate\Database\Eloquent\Collection), Object(Pterodactyl\Notifications\SendPasswordReset), Array)
#9 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Notifications/SendQueuedNotifications.php(57): Illuminate\Notifications\ChannelManager->sendNow(Object(Illuminate\Database\Eloquent\Collection), Object(Pterodactyl\Notifications\SendPasswordReset), Array)
#10 [internal function]: Illuminate\Notifications\SendQueuedNotifications->handle(Object(Illuminate\Notifications\ChannelManager))
#11 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#12 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#13 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#14 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#15 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(94): Illuminate\Container\Container->call(Array)
#16 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(114): Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}(Object(Illuminate\Notifications\SendQueuedNotifications))
#17 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Notifications\SendQueuedNotifications))
#18 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(98): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#19 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(42): Illuminate\Bus\Dispatcher->dispatchNow(Object(Illuminate\Notifications\SendQueuedNotifications), false)
#20 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(69): Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\DatabaseJob), Array)
#21 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(317): Illuminate\Queue\Jobs\Job->fire()
#22 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(267): Illuminate\Queue\Worker->process('database', Object(Illuminate\Queue\Jobs\DatabaseJob), Object(Illuminate\Queue\WorkerOptions))
#23 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(113): Illuminate\Queue\Worker->runJob(Object(Illuminate\Queue\Jobs\DatabaseJob), 'database', Object(Illuminate\Queue\WorkerOptions))
#24 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\Queue\Worker->daemon('database', 'high,standard,l...', Object(Illuminate\Queue\WorkerOptions))
#25 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(85): Illuminate\Queue\Console\WorkCommand->runWorker('database', 'high,standard,l...')
#26 [internal function]: Illuminate\Queue\Console\WorkCommand->fire()
#27 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#28 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#29 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#30 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#31 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\Container\Container->call(Array)
#32 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Command/Command.php(264): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#33 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Console/Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#34 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Application.php(869): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#35 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Application.php(223): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#36 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/symfony/console/Application.php(130): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#37 /var/www/html/mcadmin.phoenixden.net/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#38 /var/www/html/mcadmin.phoenixden.net/public_html/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#39 {main}
```

Thank you for your help in resolving this. If this is not swift mailer, please point me in the direction I should go. I have already reported to pterodactyl and they said there is nothing they can do.

Note: Each of the above errors happen at different times. Sometimes when trying to send mail it gives one error, while other times it gives the other.

**Comment by tyler2k15** [comment]
I have resolved this; posting here for anyone else with issue.

Oddly enough, the documentation I was given by one.com only gives me the port of 465 for the SMTP server. I thought maybe there was an issue with that port even though it worked on XF install so I created a test account from gmail.

I then tried the SSL port (465) and received same error, and then tried TLS (587) and that seeemed to work. So I then entered my original details in with port 587 instead (even though the documentation doesn't have it as a valid port for SMTP) and it worked!

Thank you for your help earlier
